src/poetry/mixology/version_solver.py

```diff
@@ -23,12 +24,43 @@
     from poetry.core.packages.package import Package
     from poetry.core.packages.project_package import ProjectPackage
 
+    from poetry.packages import DependencyPackage
     from poetry.puzzle.provider import Provider
 
 
 _conflict = object()
 
 
+class DependencyCache:
```

Comment:
Can/Should we move the logic into the provider itself so that the use of the cache is implicit?

Reply:
I think that this cache is specialized to the solver: really its point is to do the slightly weird thing where it filters itself on access.  I wondered about calling it a `WinnowingCache` or something like that but it's an unusual word... better ideas welcome!

So I don't think that this cache should be moved to a place where it might be mistaken for a regular cache.

Reply:
This is fine for now.